pkg/codegen/operations.go

```diff
@@ -294,6 +294,7 @@ func (o *OperationDefinition) GetResponseTypeDefinitions() ([]TypeDefinition, er
 
 					td := TypeDefinition{
 						TypeName: typeName,
+						JsonName: responseName,
```

Comment:
Using this to convey the responseName. Not sure if it is better to add a new attribute to pollute `TypeDefinition`.

Reply:
I wouldn't overload the meaning of a field. Add a ResponseName and say what its purpose is. I've found that it's better to be overly verbose in this code, it's damn complex, and this kind of field separation helps inevitable future refactoring.

Reply:
Agreed. I've added a new attribute.